tutorials/basic_1_square/README.md

```diff
@@ -0,0 +1,137 @@
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:flame/game.dart';
+
+void main() {
+  runApp(GameWidget(
+    game: MyGame(),
+  ));
+}
+
+class MyGame extends Game {
+  @override
+  void update(double dt) { /* TODO */ }
+ }
+
+  @override
+  void render(Canvas canvas) { /* TODO */ }
+}
+```
+
+That is it, you should see just an empty black screen for now, but now we have the bare bones needed to start implementing our game.
+
+Before going further, it is important to explain what those two methods mean.
+
+Flame's `Game` class is an implementation of a Game Loop, which the basic structure on which basically all games are built on. It is called a Game Loop because it really works as an infinite loop that will continue to iterate as long as the game is running. This loop can be categorized as followed:
+
+ - 1: Take input from the player
+ - 2: Process the logic for the next frame
+ - 3: Render the frame.
+
+In this tutorial we will focus on both step two and three.
+
+To process the logic of the game, we use the `update` method, which is ran always before the frame is rendered and receives a single argument, a `double` value called `dt` (delta time), which is the ammout os seconds from the current iteration and the last one. This delta time is very important so we can always know how to correct calculate movement, animations and etc.
+
+To render the frame, we use the `render` method, this method receives a single argument which is an instance of a `dart:ui` `Canvas` class. With that instance we can basically render anything we want. It is important to not that in this method, we should only contain render logic, any game logic should be put in the `update` method.
+
+Now that we have a better understanding of how the game structure works, lets start to plan our game.
+
+First things first, lets think on what variables and data structure we would need. For that, lets recap what we are building: A simple game where a square keeps bouncing forever from one side of the screen to the other. Thinking on this, we will need:
+
+ - A variable to tell us the speed of the square in logical pixels per second.
+ - A variable to inform which direction the square is moving.
+ - A structure to represent our square, which has a position and dimensions.
+
+With that in mind, check the example below, note the comments for explanations on which code section.
+
+
+```dart
+class MyGame extends Game {
+  // A constant speed, represented in logical pixels per second
+  static const int square_speed = 400;
+
+  // To represent our square we are using the Rect class from dart:ui
+  // which is a handle class to represent this type of data. We will be
+  // seeing other types of data classes in the future, but for this
+  // example, Rect will do fine for us.
+  late Rect squarePos;
```

Comment:
you could also use just a Vector2, which might be simpler to do some operations (you can have another one, static, for the dimensions)